lib/datadog.ts

```diff
@@ -8,36 +8,88 @@
 
 import * as cdk from "@aws-cdk/core";
 import * as lambda from "@aws-cdk/aws-lambda";
-import { applyLayers, redirectHandlers, addForwarder } from "./index";
+import { applyLayers, redirectHandlers, addForwarder, applyEnvVariables, defaultEnvVar } from "./index";
 
 export interface DatadogProps {
-  lambdaFunctions: lambda.Function[];
   pythonLayerVersion?: number;
   nodeLayerVersion?: number;
   addLayers?: boolean;
   forwarderARN?: string;
+  flushMetricsToLogs?: boolean;
+  site?: string;
+  apiKey?: string;
+  apiKMSKey?: string;
+  enableDDTracing?: boolean;
+  injectLogContext?: boolean;
 }
 
+export const site_list: string[] = [
+  "datadoghq.com",
+  "datadoghq.eu",
+  "us3.datadoghq.com",
+  "ddog-gov.com"
+]
+
 export class Datadog extends cdk.Construct {
-  /** allows accessing the counter function */
+  scope: cdk.Construct;
+  props: DatadogProps
   constructor(scope: cdk.Construct, id: string, props: DatadogProps) {
     super(scope, id);
-    if (props.addLayers === undefined) {
-      props.addLayers = true;
+    this.scope = scope;
+    this.props = props;
+  }
+
+  public addLambdaFunctions(lambdaFunctions: lambda.Function[]){
```

Comment:
I'm not sure what `addLambdaFunctions` is supposed to do here. It seems to do everything that the constructor should do (Setting props, choosing default values, checking for configuration errors), and then it calls `applyLayers` which seems to add the lambda functions. It might make sense to break a lot of this logic out of `addLambdaFunctions`